tests/rptest/services/redpanda.py

```diff
@@ -1066,10 +1066,25 @@ def metric_sum(self,
                 for sample in family.samples:
                     if sample.name != metric_name:
                         continue
-                    if ns and sample.labels["namespace"] != ns:
-                        continue
-                    if topic and sample.labels["topic"] != topic:
-                        continue
+                    labels = sample.labels
+                    if ns:
+                        if "redpanda_namespace" in labels:
+                            if labels["redpanda_namespace"] != ns:
+                                continue
+                        elif "namespace" in labels:
+                            if labels["namespace"] != ns:
+                                continue
+                        else:
+                            assert False, f"Missing namespace label: {sample}"
+                    if topic:
+                        if "redpanda_topic" in labels:
+                            if labels["redpanda_topic"] != topic:
+                                continue
+                        elif "topic" in labels:
+                            if labels["topic"] != topic:
+                                continue
+                        else:
+                            assert False, f"Missing topic label: {sample}"
```

Comment:
nit: these checks are a little hard to read, it looks like they could be simplified a bit and collapsed?

Reply:
Agreed it's a bit ugly. I spent some time earlier trying to clean it up, but couldn't come to anything simpler. Open to suggestions on how to improve it

Reply:
Actually maybe I'll do this in a follow up, if it works:

```
if ns:
    assert "kafka_namespace" in labels or "namespace" in labels, f"Missing namespace"
    if labels.get("kafka_namespace", labels.get("namespace")) != ns
        continue
```